stress_test_model_cb.R

```diff
@@ -489,28 +506,43 @@ for (i in seq(1, nrow(transition_scenarios))) {
 
   plan_carsten_bonds <- pacta_bonds_results %>%
     filter(
-      year == start_year,
-      technology %in% technologies,
-      scenario_geography == scenario_geography_filter
+      .data$year == .env$start_year,
+      .data$technology %in% .env$technologies,
+      .data$scenario_geography == .env$scenario_geography_filter,
+      .data$scenario %in% .env$scenario_to_follow_ls
     )
 
   financial_data_bonds_pd <- financial_data_bonds %>%
-    select(company_name, corporate_bond_ticker, ald_sector, technology, pd) %>%
-    distinct(across(everything()))
+    select(company_name, corporate_bond_ticker, ald_sector, technology, pd)
+
+  report_duplicates(
+    data = financial_data_bonds_pd,
+    cols = names(financial_data_bonds_pd)
+  )
 
+  rows_plan_carsten <- nrow(plan_carsten_bonds)
   plan_carsten_bonds <- plan_carsten_bonds %>%
-    left_join(financial_data_bonds_pd, by = c("company_name", "id" = "corporate_bond_ticker", "ald_sector", "technology"))
+    dplyr::inner_join(
+      financial_data_bonds_pd,
+      by = c("company_name", "id" = "corporate_bond_ticker", "ald_sector", "technology")
+    )
+  cat("number of rows dropped from technology_exposure by joining financial data
+      on company_name, corporate_bond_ticker, ald_sector and technology = ",
+      rows_plan_carsten - nrow(plan_carsten_bonds), "\n")
+  # TODO: ADO 879 - note which companies are removed here, due to mismatch of
+  # sector/tech in the financial data and the portfolio
   # TODO: what to do with entries that have NAs for pd?
-  # TODO: kick out NAs and record the diff
-  bonds_annual_profits <- bonds_annual_profits %>%
```

Comment:
for my understanding, why could this line be dropped?

Reply:
we `inner_join` `id` on the `corporate_bond_ticker`, which does not have any `NAs` for bonds.